### DISCUSSION TITLE
useIntersectionObserver default threshold

### DISCUSSION THREAD

**Discussion by pau-santesmasses** [vueuse/vueuse#2184] [Q&A] [answered]
Is there a good reason for the default threshold value of useIntersectionObserver to be 0.1? Since that is a percentage of the element height, large elements can behave unexpectedly, e.g. a tall div may be reported as not intersecting when it's clearly within the viewport. Overriding the threshold with a value of 0 fixes it, and it seems it would have consistent behavior regardless of the element size. It is also the default for the underlying IntersectionObserver API.

Any insight greatly appreciated!

https://github.com/vueuse/vueuse/blob/330981116f2203c9d9e62fead7cc1618cf627130/packages/core/useIntersectionObserver/index.ts#L42

**Comment by jeongtae** [comment]
I agree with this. I wasted time today, by not realizing that the default is 0.1. I was confused because there is no mention of it on the official documentation page or in the JSDoc. It was only when I looked at the source code to find the bug that I realized the default was 0.1. This should at least be stated somewhere to prevent cases like mine from happening.

**Comment by ilyaliao** [comment, marked as answer]
Solution through #4069 :)

<img width="619" alt="截圖 2025-02-08 凌晨12 54 42" src="https://github.com/user-attachments/assets/9efde978-cc76-460d-9d61-0c3e09462368" />
